app/experimenter/experiments/tests/api/v5/test_mutations.py

```diff
@@ -522,50 +581,59 @@ def test_update_experiment_publish_status(self):
         experiment = NimbusExperiment.objects.get(id=experiment.id)
         self.assertEqual(experiment.publish_status, NimbusExperiment.PublishStatus.REVIEW)
 
-    def test_end_experiment_in_kinto(self):
+    def test_update_experiment_with_invalid_publish_status_error(self):
         user_email = "user@example.com"
         experiment = NimbusExperimentFactory.create(
-            status=NimbusExperiment.Status.LIVE,
+            publish_status=NimbusExperiment.PublishStatus.REVIEW,
         )
         response = self.query(
-            END_EXPERIMENT_MUTATION,
+            UPDATE_EXPERIMENT_MUTATION,
             variables={
                 "input": {
                     "id": experiment.id,
+                    "publicDescription": "yogurt",
                 }
             },
             headers={settings.OPENIDC_EMAIL_HEADER: user_email},
         )
         self.assertEqual(response.status_code, 200, response.content)
+        content = json.loads(response.content)
+        result = content["data"]["updateExperiment"]
+        self.assertEqual(
+            result["message"],
+            {
+                "experiment": [
+                    "Nimbus Experiment has publish status Review "
+                    "but can only be changed with Idle."
+                ]
+            },
+        )
 
-        experiment = NimbusExperiment.objects.get(id=experiment.id)
-        self.assertEqual(experiment.is_end_requested, True)
-        latest_change = experiment.changes.order_by("-changed_on").first()
-        self.assertEqual(latest_change.experiment_data["is_end_requested"], True)
-
-    def test_end_experiment_in_kinto_fails_with_nonlive_status(self):
+    def test_update_publish_status_from_approved_to_review_error(self):
```

Comment:
Oh wait, hmm: I just got this failing locally, so maybe more than a flake8 thing

Reply:
Ah, I think I worked out the problem - tl;dr: multiple inheritance is tricky and I don't think these mixin classes can work this way with a shared base class.

A fix might be to extract that shared state-change validator as a standalone function and just use that in each of the mixins.

To elaborate: This is a base class for two mixins. Each of those mixins gets used as one of many base classes for the serializer.

Both of these mixins define a `FIELD` property. But, when the multiple inheritance is resolved, it looks like just one `FIELD` survives in the serializer - i.e. that of `StatusValidationMixin`. Each mixin does not have its own isolated scope for `FIELD`

So, when `PublishStatusValidationMixin` tries to validate this state transition in its base class parent `validate`, it's actually looking at `status` rather than `publish_status`.